api/v1alpha3/backstage_types.go

```diff
@@ -142,38 +141,38 @@ type ExtraFiles struct {
 	// For each item in this array, if a key is not specified, it means that all keys in the ConfigMap will be mounted as files.
 	// Otherwise, only the specified key will be mounted as a file.
 	// +optional
-	ConfigMaps []FileObjectKeyRef `json:"configMaps,omitempty"`
+	ConfigMaps []FileObjectRef `json:"configMaps,omitempty"`
 
 	// List of references to Secrets objects mounted as extra files under the MountPath specified.
 	// For each item in this array, a key must be specified that will be mounted as a file.
 	// +optional
-	Secrets []FileObjectKeyRef `json:"secrets,omitempty"`
+	Secrets []FileObjectRef `json:"secrets,omitempty"`
 
 	// List of references to Persistent Volume Claim objects mounted as extra files
 	// For each item in this array, a key must be specified that will be mounted as a file.
 	// +optional
-	Pvcs []PvcRef `json:"pvcs,omitempty"`
```

Comment:
I understand that the refactoring, but this would expose the `key` field, which IMO does not make sense on PVCs. I find it more confusing for the CR users, especially when inspecting the documentation of the fields or when using the OCP console CR editor.
![image](https://github.com/user-attachments/assets/43f1239a-0198-4007-b37e-07092b30e170)

Reply:
Also, I think keeping the `PvcRef` struct would allow later to implement an OLM descriptor on the name field, so that users can pick PVCs from an auto-populated list of PVCs - scope of https://issues.redhat.com/browse/RHIDP-3113